### PR TITLE
Handbrake => 1.8.2, update unit test workflow.

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #10340 (Specifically missing handbrake binaries as mentioned.)

- Mea culpa... I hadn't built the binaries when I updated the handbrake package file, because I was focused upon fixing builds on the Glibc 2.37 systems, and then forgot.

- This also adjusts the unit test workflow to only run the arm container if there are armv7l compatible packages changes or changes to non-package files.

##
Builds:
- [x] `x86_64`
- [ ] `i686`
- [ ] `armv7l`
##
Tested & Working properly:
- [ ] `x86_64`
- [ ] `i686`
- [ ] `armv7l` 
##
- [x] This PR has no manifest .filelist changes. _(Package changes have neither added nor removed files.)_
##
### Run the following to get this pull request's changes locally for testing.
```bash
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=handbrake crew update \
&& yes | crew upgrade
```

### CODE REVIEW

**Comment by satmandu** [comment]
@Zopolis4 I think I've changed my mind on this again. The Unit Tests should not load the armv7l container at all if no changed packages have compatibility with armv7l...

**Comment by satmandu** [comment]
Converting to draft because I'm going to use this PR to test some unit test changes..., because we do want to test non-package file updates on the armv7l containers...

File: .github/workflows/Unit-Test.yml [hunk 1]
```diff
@@ -66,6 +74,11 @@ jobs:
             esac
       - name: Run unit tests
         run: |
+             if [[ -z ${NON_PKG_CHANGED_FILES} ]] && [[ $PLATFORM == 'linux/arm/v7' ]] && [ -n "${ARMV7L_PACKAGES+1}" ]; then
```
Comment (uberhacker) [inline .github/workflows/Unit-Test.yml:77]: Won't this condition always be true?  `[ -n "${ARMV7L_PACKAGES+1}" ]`

Comment (satmandu) [reply .github/workflows/Unit-Test.yml:77]: From https://linuxsimply.com/bash-scripting-tutorial/conditional-statements/if/set-variable-if-not/

```
Similarly, you can employ the parameter expansion with the -n option to check if a variable is set. In this case, use the syntax: if [[ -n "${variable+value}" ]] where “-n” returns true only when the variable is set.
```

Comment (satmandu) [reply .github/workflows/Unit-Test.yml]: Changing the logic...

File: .github/workflows/Unit-Test.yml [hunk 2]
```diff
@@ -14,30 +14,57 @@ jobs:
...
+            export ARMV7L_PACKAGES="$(for i in ${CHANGED_PACKAGES}; do grep -q "compatibility.*all\|armv7l" packages/${i}.rb && echo ${i}; done | xargs)"
+            if [[ -n ${ARMV7L_PACKAGES} ]]; then
+              echo "ARMV7L_PACKAGES=${ARMV7L_PACKAGES}" >> $GITHUB_ENV
+              echo "PR #${{ github.event.pull_request.number }} has these armv7l compatible packages: ${ARMV7L_PACKAGES}"
+            fi
 
-            # If a package has a compatibility of 'all' or one that includes 'armv7l', add it to the list.
-            echo "PR #${{ github.event.pull_request.number }} has these armv7l compatible packages: $(for p in ${CHANGED_PACKAGES}; do grep -q "compatibility 'all'\|compatibility '.*armv7l.*'" packages/${p}.rb && echo ${p}; done | xargs)"
+            ## If a package has a compatibility of 'all' or one that includes 'i686', add it to i686_PACKAGES.
+            export i686_PACKAGES="$(for i in ${CHANGED_PACKAGES}; do grep -q "compatibility.*all\|i686" packages/${i}.rb && echo ${i}; done | xargs)"
```
Comment (Zopolis4) [inline .github/workflows/Unit-Test.yml:63]: this grep logic is broken, as i mentioned in the previous pr:
image

**Review by Zopolis4 — CHANGES_REQUESTED** [review]

File: .github/workflows/Unit-Test.yml (hunk 2 above)
Comment (satmandu) [reply .github/workflows/Unit-Test.yml:63]: Yeah agreed... Still futzing with it...

Comment (satmandu) [reply .github/workflows/Unit-Test.yml:63]: (That's why it is in draft...)

Comment (satmandu) [reply .github/workflows/Unit-Test.yml]: Good catch!

**Review by uberhacker — APPROVED** [review]
Nice work!

**Pull request merged**